src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,141 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceTest {
+    private static RegistrationService service;
+    private static StorageDao storageDao;
+    private static List<User> usersList;
+    private User alice;
+    private User bob;
+    private User david;
+
+    @BeforeAll
+    static void beforeAll() {
+        service = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+        usersList = new ArrayList<>();
+    }
+
+    @BeforeEach
+    void setUp() {
+        alice = new User("user230j", "de^wdki", 18);
+        bob = new User("user22d0j", "de^wdkHBIUi", 19);
+        david = new User("userDavid", "dewiuydki", 21);
+    }
+
+    @AfterEach
+    void tearDown() {
+        int expectedSize = usersList.size();
+        int actualSize = Storage.people.size();
+        assertEquals(expectedSize, actualSize, "Expected size should be "
+                + expectedSize + " but was: " + actualSize);
+        Storage.people.clear();
+        usersList.clear();
+    }
+
+    @Test
+    void register_validUserCase_Ok() {
+        service.register(alice);
+        usersList.add(alice);
+        service.register(bob);
+        usersList.add(bob);
+        service.register(david);
+        usersList.add(david);
+        User actual = storageDao.get(alice.getLogin());
+        assertEquals(alice, actual, "actual should be " + alice + " , but was: " + actual);
+        actual = storageDao.get(bob.getLogin());
+        assertEquals(bob, actual, "actual should be " + bob + " , but was: " + actual);
+        actual = storageDao.get(david.getLogin());
+        assertEquals(david, actual, "actual should be " + david + " , but was: " + actual);
```

Comment:
```suggestion
        User actualAlice = service.register(alice);
        User actualBob =service.register(bob);
        User actualDavid = service.register(david);
        assertEquals(alice, actualAlice, "actual should be " + alice + " , but was: " + actualAlice);
        ....
```

It's better not to use other services in unit tests.